releasenotes/notes/0.19/refactor-grover-isgoodstate-check-54fdb61f899e5158.yaml

```
@@ -0,0 +1,30 @@
+---
+features:
+  - |
+    If the optimal power of the Grover operator is not known, the :class:`~qiskit.algorithms.Grover` 
+    algorithm checks different powers (i.e. iterations) and applies the 
+    :meth:`~qiskit.algorithms.AmplificationProblem.is_good_state` function to check whether
+    a good bitstring has been measured. However, if the optimal power is known and only a single 
+    circuit is run, the ``is_good_state`` function is not required and Grover search can simply 
+    return the most likely bitstring.
+
+    In this PR, :class:`~qiskit.algorithms.Grover` and 
+    :class:`~qiskit.algorithms.AmplificationProblem` were adapted to support this behavior. For 
+    example, the following would previously throw a ``TypeError`` but is now valid:
```

I'm not sure if this is still clear, but I think the wording needs to be adjusted a bit to make it sound like a release note.

```suggestion
    When running the :class:`~qiskit.algorithms.Grover`algorithm class if the optimal power is
    known and only a single circuit is run, the :meth:`.AmplificationProblem.is_good_state` method
    is no longer required to be set and the Grover search will return the most likely bitstring.

    Generally, if the optimal power of the Grover operator is not known, the :class:`~qiskit.algorithms.Grover` 
    algorithm checks different powers (i.e. iterations) and applies the 
    :meth:`~qiskit.algorithms.AmplificationProblem.is_good_state` function to check whether
    a good bitstring has been measured.  For example, you are now able to run something like:
```

---

Done in 905141d